tests/Functional/Platform/AlterUuidColumnTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Platform;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Types\GuidType;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
+
+use function sprintf;
+
+class AlterUuidColumnTest extends FunctionalTestCase
+{
+    public const DEFAULT_UUID4 = '85626c2f-2f63-4120-b814-ecc085eaaba0';
+
+    public function testAlterToUuidColumn(): void
+    {
+        $table = static::prepareLegacyUuidTable('simple_uuid_table', 'id');
+        $table->setPrimaryKey(['id']);
+        $this->dropAndCreateTable($table);
+
+        $this->connection->executeStatement(
+            sprintf(
+                'INSERT INTO simple_uuid_table VALUES ("%s")',
+                static::DEFAULT_UUID4,
+            ),
+        );
+
+        $table->getColumn('id')
+            ->setType(Type::getType(Types::GUID));
+
+        $sm   = $this->connection->createSchemaManager();
+        $diff = $sm->createComparator()
+            ->compareTables($sm->introspectTable('simple_uuid_table'), $table);
+
+        $sm->alterTable($diff);
+
+        $table = $sm->introspectTable('simple_uuid_table');
+
+        self::assertInstanceOf(GuidType::class, $table->getColumn('id')->getType());
+
+        // Verify data integrity
+        $resultUuid = $this->connection
+            ->executeQuery('SELECT id from simple_uuid_table')
+            ->fetchOne();
+
+        static::assertEquals(static::DEFAULT_UUID4, $resultUuid);
+    }
+
+    public function testAlterUuidInForeignRelation(): void
+    {
+        $sm = $this->connection->createSchemaManager();
+
+        $parentTable = static::prepareLegacyUuidTable('parent_uuid_table', 'id');
+        $parentTable->setPrimaryKey(['id']);
+        $this->dropAndCreateTable($parentTable);
+
+        $this->connection->executeStatement(
+            sprintf(
+                'INSERT INTO parent_uuid_table VALUES ("%s")',
+                static::DEFAULT_UUID4,
+            ),
+        );
+
+        $childTable = static::prepareLegacyUuidTable('child_uuid_table', 'parent_uuid_id');
+        $childTable->addColumn('id', Types::INTEGER)->setAutoincrement(true);
+        $childTable->setPrimaryKey(['id']);
+
+        $childTable = static::prepareForeignKey($childTable, 'parent_uuid');
+        $this->dropAndCreateTable($childTable);
+
+        $this->connection->executeStatement(
+            sprintf(
+                'INSERT INTO child_uuid_table (`parent_uuid_id`) VALUES ("%s")',
+                static::DEFAULT_UUID4,
+            ),
+        );
+
+        $childTable->removeForeignKey('fk_parent_uuid_id');
```

Comment:
Temporarily removing the foreign key is kind of what I would expect the comparator to do, tbh.

Reply:
If I remove the foreign key the test will fail. The error received is:
"General error: 1833 Cannot change column 'id': used in a foreign key constraint 'fk_parent_uuid_id' of table 'doctrine_tests.child_uuid_table'"

Is it intended?

Reply:
No. This is precisely what needs to be fixed.

Reply:
I changed the test so that it must fail. I will try to allocate some time during next week in order to see how can I fix this issue. Any insight is welcome.

Reply:
After some research I figure out that in order to automatically to drop and create  the foreign keys I will need that introspectTable and hereby the Table class will contains information about which tables and and foreign keys are affected. I also tried another approach that seems to work when I run in the MySQL console and it's using compound statements (MariaDB support it), however the "executeStatement" doesn't support this type of statements.

Do you have any insight that can help?

Reply:
We should avoid compound statements and I don't think we need them.